src/core/index.ts

```diff
@@ -285,3 +287,32 @@ export abstract class CrowdinApi {
         return this.retryService.executeAsyncFunc(() => this.httpClient.patch(url, data, config));
     }
 }
+
+let deprecationEmittedForOptionalParams = false;
+
+function emitDeprecationWarning(): void {
+    if (!deprecationEmittedForOptionalParams) {
+        process.emitWarning(
+            'Passing optional parameters individually is deprecated. Pass a sole object instead',
+            'DeprecationWarning',
+        );
+        deprecationEmittedForOptionalParams = true;
+    }
+}
+
+export function isOptionalString(parameter?: string | unknown): parameter is string | undefined {
+    if (typeof parameter === 'string' || typeof parameter === 'undefined') {
+        emitDeprecationWarning();
+        return true;
+    } else {
+        return false;
+    }
+}
+
+export function isOptionalNumber(parameter?: number | unknown): parameter is number | undefined {
+    if (typeof parameter === 'number' || typeof parameter === 'undefined') {
+        return true;
```

Comment:
I believe here we also need to invoke `emitDeprecationWarning`

Reply:
Oh silly me, I forgot that